graphql/connection_test.go

```diff
@@ -517,3 +517,112 @@ func TestPaginateNodeTypeFailure(t *testing.T) {
 	}
 
 }
+
+type EmbeddedArgs struct {
+	schemabuilder.PaginationArgs
+	Additional string
+}
+
+func TestEmbeddedArgs(t *testing.T) {
+	schema := schemabuilder.NewSchema()
+	type Inner struct {
+	}
+
+	query := schema.Query()
+	query.FieldFunc("inner", func() Inner {
+		return Inner{}
+	})
+
+	inner := schema.Object("inner", Inner{})
+	item := schema.Object("item", Item{})
+	item.Key("id")
+	inner.PaginateFieldFunc("innerConnection", func(args EmbeddedArgs) ([]Item, schemabuilder.PageInfo, error) {
+		retList := make([]Item, 5)
+		retList[0] = Item{Id: 1}
+		retList[1] = Item{Id: 2}
+		retList[2] = Item{Id: 3}
+		retList[3] = Item{Id: 4}
+		retList[4] = Item{Id: 5}
+		return retList, schemabuilder.PageInfo{}, nil
+	})
+	builtSchema := schema.MustBuild()
+	q := graphql.MustParse(`
+		{
+			inner {
+				innerConnection(first: 1, after: "", additional: "jk") {
+					totalCount
+					edges {
+						node {
+							id
+						}
+						cursor
+					}
+					pageInfo {
+						hasNextPage
+						hasPrevPage
+						startCursor
+						endCursor
+					}
+				}
+			}
+	    }`, nil)
+
+	if err := graphql.PrepareQuery(builtSchema.Query, q.SelectionSet); err != nil {
+		t.Error(err)
+	}
+	e := graphql.Executor{}
+	val, err := e.Execute(context.Background(), builtSchema.Query, nil, q)
+	assert.Nil(t, err)
+
+	assert.Equal(t, map[string]interface{}{
+		"inner": map[string]interface{}{
+			"innerConnection": map[string]interface{}{
+				"totalCount": int64(5),
```

Comment:
as we talked about, i think `totalCount` is not quite right, since there might be more results that what is just passed through. What does it look like for us to support a custom resolver for `totalCount`?

Reply:
I'm leaning towards returning an integer in the resolver.